### PR TITLE
Add AutoScalingPolicy to application and tile servers

### PR DESCRIPTION
**Pull request by hectcastro**
There are several high traffic periods in the day when mapping events occur. Typically, these happen in bursts and then taper off after an hour or two. In order to protect ourselves during these surges of traffic, scale up and scale down policies have been added to the application and tile servers.

Looking at the ELB graphs for overall request count, a threshold of 60 with a period of 60 seconds and 2 consecutive evaluation periods is a good combination to trigger a scale up event. Before the following minute is over, server capacity doubles.

On the other hand, a threshold of 75 with a period of 60 seconds and 10 consecutive evaluation periods is a safe combination to trigger a scale down event. This is more conservative and protects against lulls between evaluation periods where activity drops, but then picks back up in the next minute or two.

Regardless, the desired and minimum capacity for each AutoScalingGroup remains at 2.

### CODE REVIEW

**Comment by hectcastro** [comment]
I think this is ready for review. Mostly to help protect against issues with events that have a lot of attendees or events that overlap.

/cc @kdeloach @jwalgran

File: deployment/troposphere/tiler_template.py [hunk 1]
```diff
@@ -158,6 +158,106 @@
...
+            'metricLoadBalancerName',
+            Name='LoadBalancerName',
+            Value=Ref(tile_server_load_balancer)
+        )
+    ],
+))
+
+t.add_resource(cw.Alarm(
+    'alarmTileServerRequestCountLow',
+    AlarmDescription='Application server request count low',
```
Comment (kdeloach) [inline deployment/troposphere/tiler_template.py]: Alarm description should be "Tile server".

File: deployment/troposphere/app_template.py [hunk 2]
```diff
@@ -169,6 +169,106 @@
...
+            Value=Ref(app_server_load_balancer)
+        )
+    ],
+))
+
+t.add_resource(cw.Alarm(
+    'alarmAppServerRequestCountLow',
+    AlarmDescription='Application server request count low',
+    AlarmActions=[Ref(app_server_scale_down_policy)],
+    InsufficientDataActions=[Ref(app_server_scale_down_policy)],
```
Comment (kdeloach) [inline deployment/troposphere/app_template.py]: How come the scale down policy alarms need `InsufficientDataActions` and the scale up policy alarms don't?

**Comment by kdeloach** [comment]
+1 Looks good but I'm not able to test this.

File: deployment/troposphere/app_template.py (hunk 2 above)
Comment (hectcastro) [inline deployment/troposphere/app_template.py]: If no requests come in, and we're in a scaled up state (spending more money), the scale down event does not trigger. This protects us from the scenario where a scale up event occurs followed by long periods of no traffic (no need to spend more money).

If there are `InsufficientDataActions` for `RequestCount`, then there aren't any user requests coming through the ELB. No need to scale up in that scenario.

File: deployment/troposphere/tiler_template.py (hunk 1 above)
Comment (hectcastro) [inline deployment/troposphere/tiler_template.py]: Fixed, thanks.

File: deployment/troposphere/app_template.py (hunk 2 above)
Comment (kdeloach) [inline deployment/troposphere/app_template.py]: Thanks for the explanation.

**Pull request merged**